lib/dynflow/testing/dummy_executor.rb

```diff
@@ -17,6 +17,20 @@ def delayed_event(director_event)
         @events_to_process << [director_event.execution_plan_id, director_event.step_id, director_event.event, director_event.result]
       end
 
+      def plan_events(delayed_events)
+        delayed_events.each do |event|
+          time = event.time
+          time = world.clock.current_time + time if time.is_a?(Numeric)
+          world.plan_event(event.execution_plan_id, event.step_id, event.event, time)
+        end
+      end
+
+      def execute(action, event = nil)
```

Comment:
New method called from `Testing#run_action` and `progress` as both need to process action's events